### PR TITLE
refactor(search): rating facet component refactoring

### PR DESCRIPTION
**Pull request by Sinucid**
closes: [HRZ-90315](https://spryker.atlassian.net/browse/HRZ-90315)

[HRZ-90315]: https://spryker.atlassian.net/browse/HRZ-90315?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/domain/search/facet-rating/facet-rating.component.ts [hunk 1]
```diff
@@ -1,84 +1,119 @@
...
+
+  protected override render(): TemplateResult | void {
+    const facet = this.$facet();
+
+    if (!facet) return;
+
+    return html`<oryx-search-facet-value-navigation
+      ?open=${this.open}
+      ?enableClear="${!this.disableClear}"
+      ?dirty=${!!this.$isDirty()}
```
Comment (tobi-or-not-tobi) [inline libs/domain/search/facet-rating/facet-rating.component.ts]:
```suggestion
      ?dirty=${this.$isDirty()}
```

it's a boolean already afaics

File: libs/domain/search/src/renderers/value-renderer/rating.renderer.ts [hunk 2]
```diff
@@ -14,12 +14,11 @@ export const ratingFacetRenderer = {
...
           @oryx.select=${selectListener}
           .name=${facet.name}
-          .renderLimit=${options.renderLimit}
           .open=${options.open}
           ?disableClear="${!options.enableClear}"
-          .multi=${facet.type === FacetType.Multi}
-        >
-        </oryx-search-facet-rating>
+          min="1"
+          max="4"
```
Comment (tobi-or-not-tobi) [inline libs/domain/search/src/renderers/value-renderer/rating.renderer.ts]: how come these are hardcoded?

**Review by tobi-or-not-tobi — CHANGES_REQUESTED** [review]

File: libs/domain/search/src/renderers/value-renderer/rating.renderer.ts (hunk 2 above)
Comment (Sinucid) [reply libs/domain/search/src/renderers/value-renderer/rating.renderer.ts]: I dropped default values for these props in component (to decrease amount of code and increase flexibility), facet's min and max are used by default, if values are not provided. The values are hardcoded in the renderer due to we don't have mechanism to pass per-component options on EB side for facet-navigation component, only the common set. 

For now the only way to override this hardcoded min and max - to override the renderer for the rating. But I agree, our current implementation of the facets at all are too limited

Comment (tobi-or-not-tobi) [reply libs/domain/search/src/renderers/value-renderer/rating.renderer.ts]: That's a bummer. In the former implementation, we could override the default values in the application configuration. The lack of EB is not so relevant as it's not a product that we ship anytime soon. Can we bring it back?

File: libs/domain/search/facet-price/facet-price.component.ts [hunk 3]
```diff
@@ -41,6 +43,21 @@ export class SearchPriceFacetComponent extends I18nMixin(
...
+  protected onRangeChange(e: CustomEvent<MultiRangeChangeEvent>): void {
+    const { min, max } = this.$facet()!.values;
+    const { minValue, maxValue } = e.detail;
+    const selected = {
+      min: minValue !== min ? minValue : undefined,
+      max: maxValue !== max ? maxValue : undefined,
+    } as SelectRangeFacetValues;
+
+    if (this.hasChangedValue(selected)) {
+      console.log(this.$facet(), selected);
```
Comment (tobi-or-not-tobi) [inline libs/domain/search/facet-price/facet-price.component.ts]: left over?

**Review by tobi-or-not-tobi — CHANGES_REQUESTED** [review]

File: libs/domain/search/src/renderers/value-renderer/rating.renderer.ts (hunk 2 above)
Comment (tobi-or-not-tobi) [reply libs/domain/search/src/renderers/value-renderer/rating.renderer.ts]: The idea was to allow customers to either come up with min=4 and max=4 (only showing 4-star, which is quit common) or allow to include 5

**Review by tobi-or-not-tobi — DISMISSED** [review]

**Review by tobi-or-not-tobi — APPROVED** [review]

**Pull request merged**